Add utility function to convert from Pandas dataframe to Hugging Face dataset

hickeyma opened the pull request:
Closes #190

markmc reviewed (approved):
Thanks @hickeyma !

markmc commented:
Sorry @hickeyma do you mind rebasing to commit a15c83ef56c0d4a1752392a2f316715d893da348 and also changing the new `from_pandas()` in that? Thanks!

hickeyma commented:
> Sorry @hickeyma do you mind rebasing to commit https://github.com/instructlab/sdg/commit/a15c83ef56c0d4a1752392a2f316715d893da348 and also changing the new from_pandas() in that? Thanks!

Not a bother @markmc. Rebased and updated to include utility function.
